c/collections.c

```diff
@@ -1343,11 +1350,17 @@ void *qRemove(Queue *q){
         break;
     }
   }
+
+  return currentHead;
+}
+
+void *qRemove(Queue *q) {
+
   void *result = NULL;
-  if (currentHead)
-  {
-    result = currentHead->data;
-    safeFree31((char*)currentHead,sizeof(QueueElement));
+  QueueElement *element = qDequeue(q);
+  if (element) {
+    result = element->data;
+    safeFree((char *)element, sizeof(QueueElement));
```

Comment:
Nitpick! This cast to (char *) bothers me and should be fixed in the future. It's not a breaking change and removes an irritating requirement for an unnecessary cast.

Reply:
I agree.